Source/Fuse.Reactive.JavaScript/ModuleInstance.Reactive.uno

```diff
@@ -0,0 +1,278 @@
+		}
+
+		object RemoveAt(object[] args)
+		{
+			if (args.Length == 0) throw new Error("module.removeAt(): at least one argument required");
+			else new RemoveAtOperation(this, args).JSThreadPerform();
+			return null;
+		}
+
+		object InsertAt(object[] args)
+		{
+			if (args.Length == 0) throw new Error("module.insertAt(): at least one argument required");
+			else new InsertAtOperation(this, args).JSThreadPerform();
+			return null;
+		}
+
+		abstract class Operation
+		{
+			protected readonly ModuleInstance ModuleInstance;
+			protected readonly object[] Arguments;
+			protected Operation(ModuleInstance inst, object[] args)
+			{
+				Arguments = args;
+				ModuleInstance = inst;
+			}
+
+			void UIThreadPerform()
+			{
+				var dc = (IReactive)ModuleInstance._dc;
+				dc.Set(Arguments, 0);
+			}
+
+			public void JSThreadPerform()
+			{
+				var raw = ((IRaw)ModuleInstance._dc).Raw;
+				JSThreadPerform(raw, 0);
+				UpdateManager.PostAction(UIThreadPerformStart);
+			}
+
+			void UIThreadPerformStart()
+			{
+				UIThreadPerform(ModuleInstance._dc, 0);
+			}
+
+			protected abstract int SpecialArgCount { get; }
+			protected abstract void JSThreadPerform(object dc);
+			protected abstract void UIThreadPerform(object dc);
+
+			void JSThreadPerform(object dc, int pos)
+			{
+				if (pos == Arguments.Length - SpecialArgCount)
+				{
+					JSThreadPerform(dc);
+					return;
+				}
+
+				var obj = dc as Scripting.Object;
+				if (obj != null)
+				{
+					var key = Arguments[pos].ToString();
+					JSThreadPerform(obj[key], pos+1);
+					return;
+				}
+
+				var arr = dc as Scripting.Array;
+				if (arr != null)
+				{
+					var index = Marshal.ToInt(Arguments[pos]);
+					JSThreadPerform(arr[index], pos+1);
+					return;
+				}
+
+				throw new Error("Unable to update data context. Path doesn't match exports");
+			}
+
+			void UIThreadPerform(object dc, int pos)
+			{
+				if (pos == Arguments.Length - SpecialArgCount)
+				{
+					UIThreadPerform(dc);
+					return;
+				}
+
+				var obj = dc as ObjectMirror;
+				if (obj != null)
+				{
+					var key = Arguments[pos].ToString();
+					UIThreadPerform(obj[key], pos+1);
+					return;
+				}
+
+				var arr = dc as ArrayMirror;
+				if (arr != null)
+				{
+					var index = Marshal.ToInt(Arguments[pos]);
+					UIThreadPerform(arr[index], pos+1);
+					return;
+				}
+
+				throw new Error("Unable to update data context. Path doesn't match exports");
+			}
+		}
+
+		class SetOperation: Operation
+		{
+			public SetOperation(ModuleInstance inst, object[] args): base(inst, args) {}
+
+			object _wrappedValue;
+
+			protected override int SpecialArgCount { get { return 2; } }
+			protected override void JSThreadPerform(object dc)
+			{
+				var key = Arguments[Arguments.Length-2];
+				var value = Arguments[Arguments.Length-1];
+				_wrappedValue = ModuleInstance._worker.Reflect(value);
+
+				var obj = dc as Scripting.Object;
+				if (obj != null) 
+				{
+					obj[key.ToString()] = value;
```

Comment:
I'd prefer to see a Marshal.ToType/ToString used for all conversions. There's likely some types that do not use `ToString` for conversion, if not now, then later.